gwdetchar/io/datafind.py

```diff
@@ -178,23 +172,19 @@ def get_data(channel, start, end, obs=None, frametype=None, source=None,
         series_class = TimeSeriesDict
     else:
         series_class = TimeSeries
-    # construct file cache if none is given
-    if source is None:
-        obs = obs if obs is not None else frametype[0]
-        source = gwdatafind.find_urls(obs, frametype, start, end)
     # read from frames or NDS
-    if source:
+    if source is not None:
         if isinstance(channel, (list, tuple)):
             channel = remove_missing_channels(channel, source)
         return series_class.read(
             source, channel, start=start, end=end, nproc=nproc,
             verbose=verbose, **kwargs)
     elif isinstance(channel, str):
```

Comment:
it occurs to me that this should probably be

```python
        elif not isinstance(channel, (list, tuple))
```

to match the condition above that declares `series_class`

Reply:
thanks @duncanmmacleod, I've changed this in the latest commit. I also restored the logic that attempts to load a cache using `gwdatafind` because I remembered that it's _much_ simpler that way, but provided a lightweight wrapper around `gwdatafind.find_urls` to add some protection against non-existent frame files.